ext/sodium/libsodium.c

```diff
@@ -3294,15 +3173,15 @@ PHP_FUNCTION(sodium_unpad)
 		RETURN_THROWS();
 	}
 	if (blocksize <= 0) {
-		zend_throw_exception(sodium_exception_ce, "block size cannot be less than 1", 0);
+		zend_argument_error(sodium_exception_ce, 2, "must be greater than 0");
 		RETURN_THROWS();
 	}
 	if (blocksize > SIZE_MAX) {
-		zend_throw_exception(sodium_exception_ce, "block size is too large", 0);
+		zend_argument_error(sodium_exception_ce, 2, "is too large");
 		RETURN_THROWS();
 	}
 	if (padded_len < blocksize) {
-		zend_throw_exception(sodium_exception_ce, "invalid padding", 0);
+		zend_argument_error(sodium_exception_ce, 1, "must be less than the blocksize");
```

Comment:
sorry, I've just realized this error message makes zero sense. :/  So what about something like `Argument # 1 ($string) must have a length greater than or equal to argument # 2 ($length)` instead? I couldn't come up with a better one.

Reply:
I don't think we need to force ourselves to always mention all arguments in this format. Something like `Argument #1 ($string) must not be shorter than the block size` maybe?

PS: It's annoying that the error messages in this ext aren't getting tested, so it's not visible whether the result is actually good. For example, in this case the parameter name should really be changed to `$blockSize` instead of `$length`. We haven't cared a lot about quality of parameter names in arginfo before, but now with it appearing in error messages, we may have to fix some stuff...

Reply:
> Something like Argument #1 ($string) must not be shorter than the block size maybe?

Seems good! I'll change the argument name too.

> It's annoying that the error messages in this ext aren't getting tested

:/

> now with it appearing in error messages, we may have to fix some stuff...

I agree. To make things even worse, param names in the documentation also have issues... In this case it's also `$length`, but I am sure there are a lots of inconsistencies between param names at php.net and param names in stubs. That's why I've already planned a param name "overhaul" + sync. Hopefully, I can make it if I can finish my planned RFCs in time.

Reply:
Hmm, just realized that it would be very cool to be able to generate function signatures in the documentation from the stubs.